src/date-range-input/date-range-input.tsx

```diff
@@ -275,6 +283,31 @@ export const DateRangeInput = ({
         }
     };
 
+    const handleFixedRangeDateChange = (val: string) => {
```

Comment:
the logic is very similar to `handleFixedRangeSelectionChange`, would it be possible to merge them and use one for both input and calendar selection?